File: c++/unit_test/demo/Calculator-Test.cpp
#include <gtest/gtest.h>
#include "Calculator.h"

TEST(CalculatorTest, AddNumbers) {
    EXPECT_EQ(add(2, 3), 5);
}

TEST(CalculatorTest, SubtractNumbers) {
    EXPECT_EQ(subtract(5, 3), 2);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: _experimental/_typescript/unit_test/demo/src/calculator.test.ts
import { Calculator } from './calculator';

describe('Calculator', () => {
    const calc = new Calculator();

    test('Testing add...', () => {
        expect(calc.add(10, 15)).toBe(25);
    });

    test('Testing substract...', () => {
        expect(calc.subtract(20, 5)).toBe(15);
    });
});